scylla-cdc-base/src/main/java/com/scylladb/cdc/model/master/Master.java

```diff
@@ -42,10 +46,52 @@ private GenerationId getGenerationId() throws InterruptedException, ExecutionExc
         }
     }
 
-    private boolean generationDone(GenerationMetadata generation, Set<TaskId> tasks) {
-        return generation.isClosed()
-                && !tasks.isEmpty()
-                && connectors.transport.areTasksFullyConsumedUntil(tasks, generation.getEnd().get());
+    private boolean generationDone(GenerationMetadata generation, Set<TaskId> tasks) throws ExecutionException, InterruptedException {
+        if (!generation.isClosed()) {
+            return false;
+        }
+
+        if (generationTTLExpired(generation)) {
+            return true;
+        }
+
+        return connectors.transport.areTasksFullyConsumedUntil(tasks, generation.getEnd().get());
+    }
+
+    private boolean generationTTLExpired(GenerationMetadata generation) throws ExecutionException, InterruptedException {
+        // Check the CDC tables TTL values.
+        //
+        // By default the TTL value is relatively
+        // small (24 hours), which means that we
+        // could safely skip some older generations
+        // (the changes in them have already
+        // expired).
+        Date now = Date.from(connectors.clock.instant());
```

Comment:
Why do we need clock at all? Can't we use new Date()? Is it for testing to be able to control the clock?

Reply:
This is for unit tests to be able to inject a test Date.

Reply:
https://github.com/scylladb/scylla-cdc-java/pull/19/files#diff-89f5f598b70c3f6cb07452354764082823b91c35fbe7ccfc80a6b7a69294f1b0R232